src/kvstore/raftex/RaftPart.cpp

```diff
@@ -651,49 +641,48 @@ void RaftPart::processAppendLogResponses(
             // Step 3: Commit the batch
             if (commitLogs(std::move(walIt))) {
                 committedLogId_ = lastLogId;
-
-                // Step 4: Fulfill the promise
-                if (iter.hasNonCASLogs()) {
-                    sendingPromise_.setOneSharedValue(AppendLogResult::SUCCEEDED);
-                }
-                if (iter.leadByCAS()) {
-                    sendingPromise_.setOneSingleValue(AppendLogResult::SUCCEEDED);
-                }
-                VLOG(2) << idStr_ << "Succeeded in committing the logs";
-
-                // Step 5: Check whether need to continue
-                // the log replication
-                CHECK(replicatingLogs_);
-                // Continue to process the original AppendLogsIterator if necessary
-                iter.resume();
-                if (iter.empty()) {
-                    if (logs_.size() > 0) {
-                        // continue to replicate the logs
-                        sendingPromise_ = std::move(cachingPromise_);
-                        cachingPromise_.reset();
-                        iter = AppendLogsIterator(
-                            lastLogId_ + 1,
-                            std::move(logs_),
-                            [this] (const std::string& log) -> std::string {
-                                auto res = compareAndSet(log);
-                                if (res.empty()) {
-                                    // Failed
-                                    sendingPromise_.setOneSingleValue(
-                                        AppendLogResult::E_CAS_FAILURE);
-                                }
-                                return res;
-                            });
-                        logs_.clear();
-                    } else {
-                        replicatingLogs_ = false;
-                        VLOG(2) << idStr_ << "No more log to be replicated";
-                    }
-                }
+                firstLogId = lastLogId_ + 1;
             } else {
                 LOG(FATAL) << idStr_ << "Failed to commit logs";
```

Comment:
if commitLogs() failed, why not return?

Reply:
If commitLogs failed, there must be some really horrible things happened.  To protect the data,  we must stop the server.

Reply:
LOG(FATAL) will stop storage server?